src/subtype.c

```diff
@@ -2716,6 +2741,114 @@ static jl_value_t *intersect_varargs(jl_vararg_t *vmx, jl_vararg_t *vmy, jl_sten
     return ii;
 }
 
+static int normalize_vb_offset(jl_svec_t **params JL_REQUIRE_ROOTED_SLOT, jl_tvar_t *v, jl_varbinding_t *vb, jl_vararg_t *ii, jl_stenv_t *e)
+{
+    assert(!jl_has_typevar(ii->T, v));
+    if (ii->N != (jl_value_t*)v) {
+        return 1;
+    }
+    ssize_t pos = jl_svec_len(*params) - 1;
+    if (vb->offset < 0) {
+        // Insert additional copies of the type to make up the
+        // `offset` difference.
+        *params = jl_svec_copy_resize(*params, jl_svec_len(*params) - vb->offset);
+        for (size_t i = 0; i < -vb->offset; ++i) {
+            jl_svecset(*params, pos + i, ii->T);
+        }
+        jl_svecset(*params, pos + -vb->offset, ii);
+    }
```

Comment:
There seems no reason to do this.
If `vb->offset < 0`, these parameters should have been intersected during `intersect_tuple`.
A simple example
```julia
julia> A = Tuple{Tuple{T,Vararg{T,N}},NTuple{N,T}} where {N,T}
Tuple{Tuple{T, Vararg{T, N}}, Tuple{Vararg{T, N}}} where {N, T}

julia> T = Tuple{Tuple,NTuple{4,Int}} where {M}
Tuple{Tuple, NTuple{4, Int64}}

julia> typeintersect(A, T)
Tuple{NTuple{6, Int64}, NTuple{4, Int64}}
```
The above example wont be fixed after removing this branch, but that should be a bug during re-intersection. My local log shows this PR gives wrong result within the 1st round intersection.

Reply:
Another bad regression.
```julia
julia> A = Tuple{Tuple{T,Vararg{T,N}},NTuple{N,T}} where {N,T}
Tuple{Tuple{T, Vararg{T, N}}, Tuple{Vararg{T, N}}} where {N, T}

julia> T = Tuple{Tuple,NTuple{M,Int}} where {M}
Tuple{Tuple, Tuple{Vararg{Int64, M}}} where M

julia> typeintersect(A, T)
Tuple{Tuple{Int64, Int64, Vararg{Int64, N}}, Tuple{}} where N
```
The first looks same as above.

Reply:
thanks for looking, I started this before I noticed your PR, and was looking at getting that merged instead, then to revisit whether this fixed any additional cases